Marlin/src/feature/fwretract.cpp

```diff
@@ -145,7 +145,7 @@ void FWRetract::retract(const bool retracting
     if (has_zhop && !hop_amount) {
       hop_amount += retract_zlift;                        // Carriage is raised for retraction hop
       feedrate_mm_s = planner.max_feedrate_mm_s[Z_AXIS];  // Z feedrate to max
-      current_position[Z_AXIS] -= retract_zlift;          // Pretend current pos is lower. Next move raises Z.
+      current_position[Z_AXIS] += retract_zlift;          // Pretend current pos is lower. Next move raises Z.
       SYNC_PLAN_POSITION_KINEMATIC();                     // Set the planner to the new position
```

Comment:
Explaination
FWretract trick the firmware to KEEP the current Z  alive but the carriage has moved
it save current in destination ,  adds the zlift to the current , and go to destination 
Current z value stay the same , but the carriage move
It's why here , the zlift is added to current 
Here is the recover , so , we save current to destination  ' at the beginning ' ,  and here we add zlift to current , and go to destination = carriage go down
Not present in all previous releases , just a mistake of this new merging